### PR TITLE
Updates required to work on Django 1.7

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: test_utils/__init__.py [hunk 1]
```diff
@@ -187,7 +186,7 @@ def _fixture_setup(cls):
     @classmethod
     def _fixture_teardown(cls):
         """Empty (only) the tables we loaded fixtures into, then commit."""
-        if hasattr(cls, 'fixtures') and \
+        if hasattr(cls, 'fixtures') and cls.fixtures and\
```
Comment (davedash) [inline test_utils/__init__.py:189]: Can you use parens here like you have in L175-6

File: test_utils/__init__.py [hunk 2]
```diff
@@ -326,8 +324,17 @@ def tearDownClass(cls):
...
             app_name = app_label.split('.')[-1]
             app = loading.cache.get_app(app_name)
-            del loading.cache.app_models[app_name]
-            del loading.cache.app_store[app]
+            try:
+                # Django <= 1.6.
+                del loading.cache.app_models[app_name]
+            except AttributeError:
+                # Django 1.7+.
+                del loading.cache.all_models[app_name]
```
Comment (davedash) [inline test_utils/__init__.py:332]: Wouldn't this trigger another `AttributeError`?

**Comment by davedash** [comment]
Is there a way to test this via tox/travis?

File: test_utils/__init__.py (hunk 1 above)
Comment (robhudson) [inline test_utils/__init__.py:189]: Sure. I didn't want to do too much pep8 changing but I can do a pass through.

File: test_utils/__init__.py (hunk 2 above)
Comment (robhudson) [inline test_utils/__init__.py:332]: If it's not Django 1.6 or lower it should be 1.7. If it isn't I don't know how we'd handle it. With Django's new app loading feature there may be a better way to handle this anyway but I was focused on the minimum to get things working again.

**Comment by robhudson** [comment]
> Is there a way to test this via tox/travis?

Mostly likely. For being a testing library the lack of tests is humorous. :grin:

File: test_utils/__init__.py (hunk 2 above)
Comment (davedash) [inline test_utils/__init__.py:332]: I mean what I am seeing is you try to delete `loading.cache.app_models[app_name]` and if there's an Attribute error you then try again to delete it?  I'm wondering if line 332 was a typo and you meant to delete something else.

Comment (robhudson) [inline test_utils/__init__.py:332]: Nope, one is `app_models` and one is `all_models`.

Comment (davedash) [inline test_utils/__init__.py:332]: Oh wow, sorry about that.  I suspected that it was some tiny difference.  I wish I wasn't illiterate.

# STAY IN KIDS SCHOOL

Comment (robhudson) [inline test_utils/__init__.py:332]: Thanks for the review and merge @davedash :beers:

**Pull request merged**